MapboxNavigation/NavigationMapView.swift

```diff
@@ -427,7 +439,60 @@ open class NavigationMapView: MGLMapView {
             }
         }
     }
+    /**
+     Fired when the user taps a waypoint. This recognizer fires a delegate method
+     that prompts the user to remove the waypoint from the route.
+     */
+    func didTapWaypoint(tap: UITapGestureRecognizer) {
+        guard let routes: [Route] = routes else { return }
+        let tapPoint = tap.location(in: self)
+        let tapCoordinate = self.convert(tapPoint, toCoordinateFrom: tap.view)
+        guard let candidates = waypoints(on: routes, closeTo: tapCoordinate) else { return }
+
+        //take the best one, if existant, and prompt to delete
+        guard let best = candidates.1.first, let multipoint = candidates.0.first else { return }
+        requestNew(route: multipoint, without: best)
```

Comment:
This might be a good place to slip in the delegate method. If the user implements it, give them the waypoint they selected. Otherwise, call provided boilerplate functionality.

Reply:
Should 'boilerplate functionality' just be a `navigationViewController(_:didSelectWaypoint:)`delegate method or something? Perhaps the implementer doesn't want to delete a waypoint when the user selects one...

Reply:
Right, so if they do not want this to happen, they implement `navigationViewController(_:didSelectWaypoint:)` and do something custom. Otherwise, we show the action sheet, and re-request the route.